ebean-migration/src/main/java/io/ebean/migration/MigrationConfig.java

```diff
@@ -65,7 +65,8 @@ public class MigrationConfig {
   private String basePlatform;
   private String platform;
   private Properties properties;
-
+  private boolean earlyChecksumMode;
+  private boolean autoPatchChecksum = true;
```

Comment:
I have a thought along the lines that the `<init>` migration could be used to detect if migrations have been already been run using the "legacy" mode.  That is, detect if there are "legacy" migrations based on the checksum value of the `<init>` row, when that value is `0` then the checksums should be patched (and update the `<init>` checksum to `1` to indicate that the new "early" mode is being used from now on.

```java

  // maybe change the checksum from 0 to 1 to indicate that its using the new early mode

  private MigrationMetaRow createInitMetaRow() {
  
    // change the checksum value here from 0 -> 1
    return new MigrationMetaRow(0, "I", INIT_VER_0, "<init>", 0, envUserName, runOn, 0); 
  }
```

Something like:
- `LATE` and `<init>` checksum = 0 ... means we have legacy checksums that need patching
- `LATE` and `<init>` checksum = 1  ... everything all using the new LATE mode
- `EARLY` ... just use the legacy mode

-----
I'm pondering if this could be just a single `boolean legacyMode` plus detection on the `<init>` checksum.

**_So:_**
- legacyMode = true ... just use the current "late" checksum mechanism
- legacyMode = false ... use the "early" checksum and patch the checksums if the `<init>` checksum = 0